devicetypes/smartthings/zll-white-color-temperature-bulb-4000k.src/zll-white-color-temperature-bulb-4000k.groovy

```diff
@@ -0,0 +1,182 @@
+            }
+            tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+                attributeState "level", action:"switch level.setLevel"
+            }
+            tileAttribute ("colorName", key: "SECONDARY_CONTROL") {
+                attributeState "colorName", label:'${currentValue}'
+            }
+        }
+
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+
+        controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range:"(2200..4000)") {
+            state "colorTemperature", action:"color temperature.setColorTemperature"
+        }
+        valueTile("colorTemp", "device.colorTemperature", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "colorTemperature", label: '${currentValue} K'
+        }
+
+        main(["switch"])
+        details(["switch", "colorTempSliderControl", "colorTemp", "refresh"])
+    }
+}
+
+// Globals
+private getMOVE_TO_COLOR_TEMPERATURE_COMMAND() { 0x0A }
+private getCOLOR_CONTROL_CLUSTER() { 0x0300 }
+private getATTRIBUTE_COLOR_TEMPERATURE() { 0x0007 }
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        if (event.name == "colorTemperature") {
+            event.unit = "K"
+            setGenericName(event.value)
+        }
+        sendEvent(event)
+    }
+    else {
+        log.warn "DID NOT PARSE MESSAGE for description : $description"
+        log.debug zigbee.parseDescriptionAsMap(description)
+    }
+}
+
+def off() {
+    zigbee.off() + ["delay 1500"] + zigbee.onOffRefresh()
+}
+
+def on() {
+    zigbee.on() + ["delay 1500"] + zigbee.onOffRefresh()
+}
+
+def setLevel(value) {
+    zigbee.setLevel(value) + zigbee.onOffRefresh() + zigbee.levelRefresh()
+}
+
+def refresh() {
+    def cmds = zigbee.onOffRefresh() + zigbee.levelRefresh() + zigbee.colorTemperatureRefresh()
+
+    // Do NOT config if the device is the Eaton Halo_LT01, it responds with "switch:off" to onOffConfig, and maybe other weird things with the others
+    if (!((device.getDataValue("manufacturer") == "Eaton") && (device.getDataValue("model") == "Halo_LT01"))) {
+        cmds += zigbee.onOffConfig() + zigbee.levelConfig()
+    }
+
+    cmds
+}
+
+def poll() {
+    zigbee.onOffRefresh() + zigbee.levelRefresh() + zigbee.colorTemperatureRefresh()
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    return zigbee.levelRefresh()
+}
+
+def healthPoll() {
+    log.debug "healthPoll()"
+    def cmds = poll()
+    cmds.each{ sendHubCommand(new physicalgraph.device.HubAction(it))}
```

Comment:
If you like, you can wrap commands using response() to create Hub commands (instead of new physicalgraph.device.HubAction) or use sendHubCommand(List actions, delayInBetween = 1000) with default delays and use response() to create HubMultiAction